pkg/infra/iac2/templates_compiler.go

```diff
@@ -263,7 +279,7 @@ func (tc TemplatesCompiler) resolveDependencies(resource core.Resource) string {
 }
 
 // resolveStructInput translates a value to a form suitable to inject into the typescript as an input to a function.
-func (tc TemplatesCompiler) resolveStructInput(resourceVal *reflect.Value, childVal reflect.Value, useDoubleQuotedStrings bool, iacTag string) (string, error) {
+func (tc TemplatesCompiler) resolveStructInput(resourceVal *reflect.Value, childVal reflect.Value, useDoubleQuotedStrings bool, iacTag string, appliedOutputs *[]AppliedOutput) (string, error) {
```

Comment:
```suggestion
func (tc TemplatesCompiler) resolveStructInput(resourceVal *reflect.Value, childVal reflect.Value, useDoubleQuotedStrings bool, iacTag string, appliedOutputs []AppliedOutput) (string, error) {
```

Arrays/slices are already pointers.

Reply:
will update

Reply:
> Arrays/slices are already pointers.

This approach affects the underlying var from the call site, which wouldn't happen without the `*`.

```go
func TestAppend(t *testing.T) {
	t.Run("pointer to array", func(t *testing.T) {
		var slice []string
		appendHello(&slice)
		assert.Equal(t, []string{"hello"}, slice)
	})

	t.Run("array itself", func(t *testing.T) {
		var slice []string
		appendWorld(slice)
		assert.Equal(t, []string{"world"}, slice)
	})
}

func appendHello(strs *[]string) {
	*strs = append(*strs, "hello")
}

func appendWorld(strs []string) {
	strs = append(strs, "world")
}

```

`pointer_to_array` succeeds, `array_itself` fails.

Reply:
actually i tried that and it didnt persist the data to the one

Reply:
Oh, I see... I didn't trace it all the way down. It's a little weird, possibly worth a wrapper type but good for now 👍